ethdb/memory_database.go

```diff
@@ -58,15 +59,27 @@ func (db *BoltDatabase) MemCopy() Database {
 	if err != nil {
 		panic(err)
 	}
-	if err := mem.Update(func(memTx *bolt.Tx) error {
-		return db.db.View(func(tx *bolt.Tx) error {
-			return tx.ForEach(func(name []byte, b *bolt.Bucket) error {
-				memB, err := memTx.CreateBucket(name, true)
-				if err != nil {
-					return err
+
+	if err := mem.Batch(func(writeTx *bolt.Tx) error {
```

Comment:
Will be any difference between Batch and Update here? Because `mem` is created fresh in the beginning of the function, I would not expect any concurrent modifications at this point, no?